src/actions/receive-profile.js

```diff
@@ -658,39 +658,69 @@ function _fileReader(input) {
   };
 }
 
+/**
+ * Multiple file formats are supported. Look at the file type and try and
+ * parse the contents according to its type.
+ */
 export function retrieveProfileFromFile(
-  file: File
+  file: File,
+  // Allow tests to inject a custom file reader to bypass the DOM APIs.
+  fileReader: typeof _fileReader = _fileReader
 ): ThunkAction<Promise<void>> {
   return async dispatch => {
+    // Notify the UI that we are loading and parsing a profile. This can take
+    // a little bit of time.
     dispatch(waitingForProfileFromFile());
 
-    try {
-      const text = await _fileReader(file).asText();
-      const profile = unserializeProfileOfArbitraryFormat(text);
-      if (profile === undefined) {
-        throw new Error('Unable to parse the profile.');
-      }
-
-      dispatch(viewProfile(profile));
-      return;
-    } catch (e) {
-      // continuing the function normally, as we return in the try block above;
-    }
+    switch (file.type) {
+      case 'application/json':
+        // Parse JSON serialized profiles.
+        try {
+          const text = await fileReader(file).asText();
+          const profile = unserializeProfileOfArbitraryFormat(text);
+          if (profile === undefined) {
+            throw new Error('Unable to parse the profile.');
+          }
 
-    try {
-      const buffer = await _fileReader(file).asArrayBuffer();
-      const arrayBuffer = new Uint8Array(buffer);
-      const decompressedArrayBuffer = await decompress(arrayBuffer);
-      const textDecoder = new TextDecoder();
-      const text = await textDecoder.decode(decompressedArrayBuffer);
-      const profile = unserializeProfileOfArbitraryFormat(text);
-      if (profile === undefined) {
-        throw new Error('Unable to parse the profile.');
-      }
+          dispatch(viewProfile(profile));
+        } catch (error) {
+          dispatch(errorReceivingProfileFromFile(error));
+        }
+        break;
+      case 'application/x-gzip':
```

Comment:
on linux, I have `application/gzip` here, so I have an error when I try to load my gzipped profile. I guess that's why the previous code didn't try to sniff :-) Maybe on Windows we have something even different, or maybe different depending on the browser too... (octet-stream ? ;) )

I think we can simply add my `'application/gzip'` and see if anybody complains about another mime type.